core/codegen/builder.go

```diff
@@ -66,7 +66,51 @@ func (b *Builder) CallIndirect(f *Value, args ...*Value) *Value {
 	return b.call(f.llvm, fty.Signature, f.name, args)
 }
 
-func (b *Builder) call(v llvm.Value, sig Signature, name string, args []*Value) *Value {
+func (b *Builder) call(fn llvm.Value, sig Signature, name string, args []*Value) *Value {
+	l := b.callArgs(sig, name, args)
+	if sig.Result == b.m.Types.Void {
+		name = ""
+	}
+	return b.val(sig.Result, b.llvm.CreateCall(fn, l, name))
+}
+
+// Invoke invokes the function f with the specified arguments.
+// If an exception is thrown while calling f, then cleanup will be called before
+// rethrowing the exception.
+func (b *Builder) Invoke(f *Function, cleanup func(), args ...*Value) *Value {
```

Comment:
Are we using exceptions here to just plumb through errors? 
I.e. so we don't have to return true/false from every method (slice access etc).

Reply:
Yes - these are for `abort`s and `assert`s. The exceptions here are 'zero cost', and increse performance by about 10% - 15% based on my measurements.